File: platform/src/main/java/com/nixvision/support/terminal/platform/app/domains/core/source/dbs/repositories/DbConnectionRepository.java
package com.nixvision.support.terminal.platform.app.domains.core.source.dbs.repositories;


import com.nixvision.support.terminal.platform.app.domains.core.source.dbs.entities.DbConnection;
import lombok.RequiredArgsConstructor;
import org.dizitart.no2.objects.ObjectRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

import static org.dizitart.no2.objects.filters.ObjectFilters.eq;

@Repository
@RequiredArgsConstructor
public class DbConnectionRepository{


    private final ObjectRepository<DbConnection> dbConnectionRepository;

    public DbConnection add(DbConnection dbConnection) {
        dbConnectionRepository.insert(dbConnection);
        return dbConnectionRepository.find(eq("id", dbConnection.getId())).firstOrDefault();
    }

    public DbConnection update(DbConnection dbConnection) {
        dbConnectionRepository.update(dbConnection);
        return dbConnectionRepository.find(eq("id", dbConnection.getId())).firstOrDefault();
    }


    public DbConnection findById(String id) {
        return dbConnectionRepository.find(eq("id", id)).firstOrDefault();
    }

    public List<DbConnection> findAll() {
        return dbConnectionRepository.find().toList();
    }

    public void deleteById(String id) {
        dbConnectionRepository.remove(eq("id", id));
    }
}

File: src/main/kotlin/io/github/support/terminal/application/domains/db_connection/repository/DbConnectionRepository.kt
package io.github.support.terminal.application.domains.db_connection.repository

import org.dizitart.no2.objects.ObjectRepository
import org.dizitart.no2.objects.filters.ObjectFilters
import org.springframework.stereotype.Service

@Service
class DbConnectionRepository(
        val db: ObjectRepository<DbConnectionDetails>
) {

    fun add(dbConnection: DbConnectionDetails) {
        db.insert(dbConnection)
    }

    fun update(dbConnection: DbConnectionDetails) {
        db.update(dbConnection)
    }

    fun findById(id: String): DbConnectionDetails? {
        return db.find(ObjectFilters.eq("id", id)).firstOrNull()
    }

    fun findAll(): List<DbConnectionDetails> {
        return db.find().toList()
    }

    fun deleteById(id: String) {
        db.remove(ObjectFilters.eq("id", id))
    }
}